ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/MediaPlayerService.kt

```diff
@@ -62,20 +69,25 @@ class MediaPlayerService : Service() {
     private val mediaPlayerLifecycleSupport by inject<MediaPlayerLifecycleSupport>()
 
     private var mediaSession: MediaSessionCompat? = null
-    private var mediaSessionToken: MediaSessionCompat.Token? = null
     private var isInForeground = false
     private var notificationBuilder: NotificationCompat.Builder? = null
 
+    val executorService: ExecutorService = Executors.newFixedThreadPool(4)
+
+    private val MEDIA_BROWSER_ROOT_ID = "_Ultrasonice_mb_root_"
+    private val MEDIA_BROWSER_ALBUM_LIST_ROOT = "_Ultrasonic_mb_album_list_root_"
+    private val MEDIA_BROWSER_ALBUM_PREFIX = "_Ultrasonic_mb_album_prefix_"
+    private val MEDIA_BROWSER_EXTRA_ENTRY_BYTES = "_Ultrasonic_mb_extra_entry_bytes_"
+
     private val repeatMode: RepeatMode
         get() = Util.getRepeatMode()
 
-    override fun onBind(intent: Intent): IBinder {
-        return binder
-    }
-
     override fun onCreate() {
         super.onCreate()
 
+        updateMediaSession(null, PlayerState.IDLE)
+        mediaSession!!.isActive = true
```

Comment:
Previously this service delayed creating of the Media Session for as long as possible. For Android Auto we need to create it as soon as possible, inside the `onCreate`. The doc [here](https://developer.android.com/training/cars/media#browser_workflow) explains this. Thankfully the existing function works well, except IDLE sets the `mediaSession.isActive` to false.

Reply:
And for Auto you need it to be active?
If I remember correctly, if it is active it will show lockscreen controls, which we wanted to avoid as lo g as we don't actively play something

Reply:
Looks like it's not needed. This one must have been an artifact of my previous flailing, which I did a lot of.